File: leetcode/queue_stack/683_K_Empty_Slots/codes/sliding.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int kEmptySlots(int* flowers, int flowersSize, int k) {
    int *days = (int*)malloc(sizeof(int) * flowersSize);
    for (int i = 0; i < flowersSize; i++) {
        days[flowers[i] - 1] = i + 1;
    }

    int left = 0, right = k + 1;
    int res = INT_MAX;

    for (int i = 0; right < flowersSize; i++) {
        if (days[i] < days[left] || days[i] <= days[right]) {
            if (i == right) {
                int maxDay = days[left] > days[right] ? days[left] : days[right];
                if (maxDay < res) res = maxDay;
            }
            left = i;
            right = i + k + 1;
        }
    }

    free(days);
    return (res == INT_MAX) ? -1 : res;
}

File: leetcode/queue_stack/683_K_Empty_Slots/codes/sliding.py
from typing import List

class Solution:
    def kEmptySlots(self, flowers: List[int], k: int) -> int:
        n = len(flowers)
        days = [0] * n
        for i, pos in enumerate(flowers):
            days[pos - 1] = i + 1

        left, right = 0, k + 1
        res = float('inf')

        for i in range(n):
            if right >= n:
                break
            if days[i] < days[left] or days[i] <= days[right]:
                if i == right:
                    res = min(res, max(days[left], days[right]))
                left = i
                right = i + k + 1

        return -1 if res == float('inf') else res
